app/test_main.py

```diff
@@ -2,0 +1,22 @@
+from unittest import mock
+from app.main import can_access_google_page
+
+
+def test_can_access_google_page_with_internet_check():
+    with mock.patch("app.main.valid_google_url") as mock_valid, \
+            mock.patch("app.main.has_internet_connection") as mock_internet:
+        mock_valid.return_value = True
+        mock_internet.return_value = True
+
+        result = can_access_google_page("https://mate.academy/")
+        assert result == "Accessible"
+
+
+def test_can_not_access_google_page_with_internet_check():
```

Comment:
The test function name `test_can_not_access_google_page_with_internet_check` could be more descriptive. Consider renaming it to something like `test_not_accessible_when_url_valid_but_no_internet` to better reflect the conditions being tested, as per the checklist recommendation.